pkg/io/binaryReader.go

```diff
@@ -21,6 +21,8 @@ type BinReader struct {
 	u16 []byte
 	u8  []byte
 	Err error
+
+	Context interface{}
```

Comment:
I don't think it's a good idea. You can keep everything you need in the structure itself of in some kind of lambda.

Reply:
Fixed.